docs/2019-03-28-All-Path-Reachability-Proofs.md

```diff
@@ -0,0 +1,253 @@
+Proving All-Path Reachability Claims
+====================================
+
+This document details All-Path Reachability without solving the
+most-general-unifier (MGU) problem.
+MGU will be detailed in a separate document.
+
+Background
+----------
+
+Recommended reading: [proving One-Path Reachability
+claims](2018-11-08-One-Path-Reachability-Proofs.md).
+
+Similarly to the One-Path Reachability document, we assume all pattern variables
+used in this document to be _extended function-like patterns_, that is patterns
+which can be written as `t ∧ p` where the interpretation of `t` contains at most
+one element and `p` is a predicate.
+
+_Extended constructor patterns_ will be those extended function-like patterns
+for which `t` is a functional term, composed out of constructor-like symbols
+and variables.
+
+
+Definitions
+-----------
+
+### Weak always
+
+Given a formula `φ`, let `[w]φ` denote the formula “weak always” `φ`,
+defined by:
+
+```
+[w]φ := νX.φ ∨ (○X ∧ •⊤)
```

Comment:
What is the role/purpose of `∧ •⊤`?

Reply:
Intuitively, it is to make sure that when `φ` doesn't hold at current state, *then the current state doesn't get stuck*, and [w]φ holds on all next states.

Reply:
I added a explanatory paragraph concerning this.  Thanks @ttuegel and @hreada